packages/make-styles-webpack-loader/src/transformSync.ts

```diff
@@ -0,0 +1,47 @@
+import * as Babel from '@babel/core';
+import babelPluginMakeStyles from '@fluentui/babel-make-styles';
+
+export type TransformOptions = {
+  filename: string;
+
+  inputSourceMap: Babel.TransformOptions['inputSourceMap'];
+  enableSourceMaps: boolean;
+};
+
+export type TransformResult = {
+  code: string;
+  sourceMap: NonNullable<Babel.BabelFileResult['map']> | undefined;
+};
+
+/**
+ * Transforms passed source code with Babel, uses user's config for parsing, but ignores it for transforms.
+ */
+export function transformSync(sourceCode: string, options: TransformOptions): TransformResult {
+  // Parse the code first so Babel will use user's babel config for parsing
+  // During transforms we don't want to use user's config
+  const ast = Babel.parseSync(sourceCode, {
+    caller: { name: 'make-styles' },
+
+    filename: options.filename,
+    inputSourceMap: options.inputSourceMap,
+    sourceMaps: options.enableSourceMaps,
+  })!;
+
+  const babelFileResult = Babel.transformFromAstSync(ast, sourceCode, {
+    // Ignore all user's configs and apply only our plugin
+    babelrc: false,
+    configFile: false,
+    plugins: [[babelPluginMakeStyles]],
+
+    filename: options.filename,
+
+    sourceMaps: options.enableSourceMaps,
+    sourceFileName: options.filename,
+    inputSourceMap: options.inputSourceMap,
+  })!;
+
+  return {
+    code: babelFileResult.code as string,
```

Comment:
is this override safe? again I'd say we should rather implement proper behaviours

`code: babelFileResult.code ?? '',`

Reply:
> is this override safe? again I'd say we should rather implement proper behaviours

Yes, as we use code generation (it's enabled by default) we will always get `code` in output:
- https://github.com/babel/babel/blob/672a58660f0b15691c44582f1f3fdcdac0fa0d2f/packages/babel-core/src/transformation/index.ts#L53-L57
- https://github.com/babel/babel/blob/672a58660f0b15691c44582f1f3fdcdac0fa0d2f/packages/babel-core/src/transformation/file/generate.ts#L9-L15

Reply:
hmm I'd still prefer to try to follow type flow although it's covered in implementation...
anyways feel free to keep as is, thx